staging/src/k8s.io/api/storage/v1alpha1/types.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeAttachment captures the intent to attach or detach the specified volume
+// to/from the specified node.
+//
+// VolumeAttachment objects are non-namespaced.
+type VolumeAttachment struct {
```

Comment:
I am not sure what's the process here, should we add also StorageClass to v1alpha1 api? It looks like we're removing it.

Reply:
Nope according to @lavalamp we just need to introduce the new VolumeAttachment object in v1alpha1 we can leave the StorageClass objects in v1beta1 and v1.

Reply:
Seems has some relationship with https://github.com/kubernetes/kubernetes/pull/54463#issuecomment-345605558